dts/arm/nrf51_pca10028.dts

```diff
@@ -23,3 +23,42 @@
 	current-speed = <115200>;
 	status = "ok";
 };
+
+&flash0 {
+	/*
+	 * If chosen's zephyr,code-partition
+	 * is unset, the image will be linked
+	 * into the entire flash device.  If
+	 * it points to an individual
+	 * partition, the code will be linked
+	 * to, and restricted to that
+	 * partition.
+	 */
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x8000>;
+		};
+		slot0_partition: partition@8000 {
+			label = "image-0";
+			reg = <0x00008000 0x1a000>;
+		};
+		slot1_partition: partition@22000 {
+			label = "image-1";
+			reg = <0x00022000 0x1a000>;
+		};
+		scratch_partition: partition@3c000 {
+			label = "image-scratch";
+			reg = <0x0003c000 0x2000>;
+		};
+
+		/*
+		 * The flash starting at 0x0003e000 and ending at
+		 * 0x0003ffff is reserved for use by the application.
+		 */
```

Comment:
@mbolivar I don't know about sectors, so I didn't add the info here

Reply:
No issues from me.

Reply:
What I imagine is that this can store a status or downloaded image map etc. So I think it should be proportional to the image bank size. So maybe reduce this to 4 KB (so 4 sectors) as it is 12 KB for nRF52840.

Reply:
@mbolivar is this the portion of the flash used by a potential filesystem? That's what I understand at least

Reply:
Could be for a filesystem, but it doesn't have to be. E.g. we are currently prototyping LWM2M credential storage using "raw" flash_read() on partition that has a known layout.

The entire thing is probably going to need to be revisited when file systems really start to see some use.